### ISSUE TITLE
[Bug/Feature request?] Always show tab bar

### ISSUE THREAD

**Issue by kokoshneta** [TablePlus/TablePlus#3005]
### Please answer the questions below, it helps us to keep track of the issue. Please do not ignore it or your issue will be closed.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
_Microsoft SQL Server 2019; MariaDB 10.3.27 (presumably driver-agnostic)_

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
_5.3.6 (496), macOS (probably version-agnostic?)_

3. The steps to reproduce this issue:
_Open a connection, then open one or more tables in tabs_

---

I’m not sure if this is a bug or expected behaviour. If it’s expected behaviour, then consider this a feature request.

When you have multiple tables open in tabs in TablePlus, a tab bar appears to show you an overview, as in most browsers. If you only have a single table open, however, this tab bar is **always** hidden; I at least cannot find any way to show it.

This is quite annoying, because the only way to tell whether a tab is reusable (unmodified) or non-reusable (pinned or modified), – i.e., whether opening another table will reuse the same tab or create a new one – is to look at whether or not the table name is in italics _in the tab bar_.

So if you only have one tab open, there is _no way to know_ if opening another table will reuse the current tab or not. Short of actually modifying data, there is also, as far as I can tell, no way to pin the current tab if it’s the only one open, since pinning is done by double-clicking the tab _in the tab bar_.

In most browsers that have tabs, there is an option named “Always show tab bar” or similar, which makes sure that the tab bar is _always_ shown, even if there’s only one tab in the current window. Emulating that in TablePlus would be very useful.

(The “Show/hide tab bar” option in the _View_ menu would be an expected option to control this, but toggling it seems to have no discernible effect on anything at all, regardless of the number of tabs open. If I open another connection in the same window, I get the vertical tabs on the left side, and then the option is force-set to “Hide tab bar”, regardless of its previous setting, and is greyed out. Perhaps it only refers to that side tab bar, then, and not to the horizontal tab bar… but it seems perverse to make a setting that you can only control when it has no effect. Basically, I have no idea what that option is supposed to do.)

**Comment by lllopo** [comment]
I second this. In addition - there is no option to turn off (or at least i can not find it) the so-called re-usable tabs. The combination of these tow is super annoying and drives me crazy - every time I try to open a new table it re-uses the one I'm currently working on. Please, admit some (I'd say - many) people do not need re-usable tabs at all. So, both options badly needed.

**Comment by huyphams** [comment]
Hi @lllopo you can:
- double click the item to keep the tab.
- double click the tab title to keep the tab.
- Making any change or doing any filter to keep the tab.
- Use middle mouse.
- Right click > open in new tab...
.......

**Comment by lllopo** [comment]
@huyphams yes, I know. Doesn't change a thing. This behaviour is counter-intuitive and not in line with any other dev tool I use. Intellij IDEA, VS Code and all others that have re-usable (or quick view or whatever you want to call it) tabs have the option to turn **off** this unnecessary to many feature. 
I simply don't want to waste reflexes to think about what kind of tab I've opened when I work but just a simple plain consistent behaviour. No matter how you put - it is super annoying, I've saw another (locked for some reason) issue where many others complain of the same thing, so it's definitely not just me.
I really don't get it why the resistance to make 'single click on/off' a setting. 
But anyway - let's not go too much off-topic here - showing tab bar when  only one tab is open is also necessary in some cases (like sometimes I'd like to just cleanup closing all tabs and this would be an easy way to do without Command-W) plus it would mitigate a tiny bit the quick view tabs thing.

**Comment by jason3w** [comment]
Agreed. Makes closing all tabs harder. Would be great to have the option to always show the tab bar.

**Comment by JonathanGetahun** [comment]
+1

**Comment by JonathanGetahun** [comment]
+1

**Comment by huyphams** [comment]
Will be changed in the up coming version 7.0

See details here:

https://x.com/TablePlus/status/1889596044767305878
